utils/post.js

```diff
@@ -334,30 +334,27 @@ const modifyNewItemAnonymity = (newItem) => {
 };
 
 async function filterFeeds(userId, feeds = [], feed_id = null, threshold = null) {
-  const results = await Promise.all(
-    feeds.map(async (item) => {
-      const expired = isExpiredPost(item);
+  const newResult = feeds
+    .filter(async (item) => {
+      const expired = await isExpiredPost(item);
       if (expired) {
         deleteExpiredPostFromFeed(item, feed_id);
-        return null;
-      }
-      if (threshold && (item.final_score || 0) < threshold) {
-        return null;
+        return false;
       }
+      return !(expired || (threshold && (item.final_score || 0) < threshold));
+    })
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The use of `async` inside the `filter` function is incorrect. The `filter` function expects a synchronous function and will not wait for promises to resolve. This means that your filter operation will always return true because a Promise object is always truthy, and it won't actually filter out any items based on the asynchronous condition.

Instead, you should first map over the array and perform all asynchronous operations, then filter the results. Here's how you can do it:

```diff
-  const newResult = feeds
-    .filter(async (item) => {
-      const expired = await isExpiredPost(item);
-       if (expired) {
-         deleteExpiredPostFromFeed(item, feed_id);
-        return false;
-       }
-      return !(expired || (threshold && (item.final_score || 0) < threshold));
-    })
-    .map(async (item) => {
-       let newItem = {...item};
- 
-       newItem = modifyNewItemAnonymity(newItem);
-       newItem = modifyReactionsPost(newItem, newItem.anonimity);
-       const isValidPollPost = item.verb === POST_VERB_POLL && item?.polls?.length > 0;
-       if (isValidPollPost) newItem = await modifyPollPostObject(userId, newItem);
-       else newItem = await modifyPostLinkPost(DomainPage, newItem);
- 
-       return newItem;
-    });
+  const newResult = await Promise.all(feeds.map(async (item) => {
+    const expired = await isExpiredPost(item);
+    if (expired) {
+      deleteExpiredPostFromFeed(item, feed_id);
+      return null;
+    }
+    if (threshold && (item.final_score || 0) < threshold) {
+      return null;
+    }
+    let newItem = {...item};
+
+    newItem = modifyNewItemAnonymity(newItem);
+    newItem = modifyReactionsPost(newItem, newItem.anonimity);
+    const isValidPollPost = item.verb === POST_VERB_POLL && item?.polls?.length > 0;
+    if (isValidPollPost) newItem = await modifyPollPostObject(userId, newItem);
+    else newItem = await modifyPostLinkPost(DomainPage, newItem);
+
+    return newItem;
+  })).filter(item => item !== null);
```


<!-- This is an auto-generated comment by OSS CodeRabbit -->